modules/prediction/evaluator/evaluator_manager.cc

```diff
@@ -187,6 +212,60 @@ void EvaluatorManager::EvaluateObstacle(Obstacle* obstacle) {
   EvaluateObstacle(obstacle, dummy_dynamic_env);
 }
 
+void EvaluatorManager::BuildCurrentFrameEnv() {
+  auto obstacles_container =
+      ContainerManager::Instance()->GetContainer<ObstaclesContainer>(
+          AdapterConfig::PERCEPTION_OBSTACLES);
+  CHECK_NOTNULL(obstacles_container);
+  auto ego_pose_container =
+      ContainerManager::Instance()->GetContainer<PoseContainer>(
+          AdapterConfig::LOCALIZATION);
+  CHECK_NOTNULL(ego_pose_container);
+  FrameEnv curr_frame_env;
+  curr_frame_env.set_timestamp(obstacles_container->timestamp());
+  std::vector<int> obstacle_ids =
+      obstacles_container->curr_frame_predictable_obstacle_ids();
+  obstacle_ids.push_back(-1);
+  for (int id : obstacle_ids) {
+    Obstacle* obstacle = obstacles_container->GetObstacle(id);
+    if (obstacle == nullptr || obstacle->history_size() == 0) {
+      continue;
+    }
+    size_t num_frames = std::min(static_cast<size_t>(10),
+                                 obstacle->history_size());
+    ObstacleHistory obstacle_history;
+    for (size_t i = 0; i < num_frames; ++i) {
+      const Feature& obstacle_feature = obstacle->feature(i);
+      Feature feature;
+      feature.set_id(obstacle_feature.id());
+      feature.set_timestamp(obstacle_feature.timestamp());
+      feature.mutable_position()->CopyFrom(obstacle_feature.position());
+      feature.set_theta(obstacle_feature.velocity_heading());
+      if (obstacle_feature.id() != -1) {
+        feature.mutable_polygon_point()->CopyFrom(
+            obstacle_feature.polygon_point());
+        feature.set_length(obstacle_feature.length());
+        feature.set_width(obstacle_feature.width());
+      } else {
+        feature.set_length(5.0);
+        feature.set_width(2.1);
```

Comment:
Are these vehicle parameters? We should get from vehicle config to avoid hard code the parameters.

Reply:
Will fix it in the next commit, Thanks.